### PR TITLE
Add project detail and unit tests

### PR DESCRIPTION
**Pull request by marshmalien**
##### SUMMARY
* Add project detail view
* Add project detail unit tests
* Extend project type checking 

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2019-10-09 at 11 03 25 AM

### CODE REVIEW

File: awx/ui_next/src/screens/Project/ProjectDetail/ProjectDetail.jsx [hunk 1]
```diff
@@ -1,10 +1,155 @@
...
+      }`;
+    } else {
+      createdBy = formatDateString(created);
+    }
+  }
+
+  let modifiedBy = '';
+  if (modified) {
+    if (summary_fields.modified_by && summary_fields.modified_by.username) {
+      modifiedBy = `${formatDateString(modified)} ${i18n._(t`by`)} ${
```
Comment (mabashian) [inline awx/ui_next/src/screens/Project/ProjectDetail/ProjectDetail.jsx]: This is going to be nit-picky but can we go ahead and wrap the entire thing (as opposed to just `by`) for translation like https://github.com/ansible/awx/pull/4915/files#diff-41634e48aece6599257ec1cf59a0cbb8R111-R114

That lets the translator re-arrange the order of elements depending on the langauage

**Review by mabashian — CHANGES_REQUESTED** [review]
Just left one small comment - otherwise this looks great!

**Comment by ryanpetrello** [comment]
@marshmalien could you rebase this PR against `devel` to get CI passing?

**Review by mabashian — APPROVED** [review]
🎉

**Pull request merged**